crates/router/src/core/user.rs

```diff
@@ -2760,3 +2760,425 @@ pub async fn list_profiles_for_user_in_org_and_merchant_account(
+    request: user_api::SwitchOrganizationRequest,
+    user_from_token: auth::UserFromToken,
+) -> UserResponse<user_api::TokenResponse> {
+    if user_from_token.org_id == request.org_id {
+        return Err(UserErrors::InvalidRoleOperationWithMessage(
+            "User switching to same org".to_string(),
+        )
+        .into());
+    }
+
+    let user = user_from_token.get_user_from_db(&state).await?;
+    let key_manager_state = &(&state).into();
+    let role_info = roles::RoleInfo::from_role_id(
+        &state,
+        &user_from_token.role_id,
+        &user_from_token.merchant_id,
+        &user_from_token.org_id,
+    )
+    .await
+    .change_context(UserErrors::InternalServerError)?;
+
+    if role_info.get_entity_type() == EntityType::Internal {
+        return Err(UserErrors::InvalidRoleOperationWithMessage(
+            "Org switching not allowed for Internal role".to_string(),
+        )
+        .into());
+    }
+
+    let user_role = state
+        .store
+        .list_user_roles(
+            &user_from_token.user_id,
+            Some(&request.org_id),
+            None,
+            None,
+            None,
+            None,
+        )
+        .await
+        .change_context(UserErrors::InternalServerError)?
+        .into_iter()
+        .find(|role| role.status == UserStatus::Active)
+        .ok_or(UserErrors::InvalidRoleOperationWithMessage(
+            "No user role found for the requested org id".to_string(),
+        ))?
+        .to_owned();
+
+    let merchant_id = if let Some(merchant_id) = &user_role.merchant_id {
+        merchant_id.clone()
+    } else {
+        state
+            .store
+            .list_merchant_accounts_by_organization_id(
+                key_manager_state,
+                request.org_id.get_string_repr(),
+            )
+            .await
+            .change_context(UserErrors::InternalServerError)?
+            .first()
+            .ok_or(UserErrors::InternalServerError)?
+            .get_id()
+            .clone()
+    };
+
+    let profile_id = if let Some(profile_id) = &user_role.profile_id {
+        profile_id.clone()
+    } else {
+        let merchant_key_store = state
+            .store
+            .get_merchant_key_store_by_merchant_id(
+                key_manager_state,
+                &merchant_id,
+                &state.store.get_master_key().to_vec().into(),
+            )
+            .await
+            .change_context(UserErrors::InternalServerError)?;
+
+        state
+            .store
+            .list_business_profile_by_merchant_id(
+                key_manager_state,
+                &merchant_key_store,
+                &merchant_id,
+            )
+            .await
+            .change_context(UserErrors::InternalServerError)?
+            .first()
+            .ok_or(UserErrors::InternalServerError)?
+            .profile_id
+            .clone()
+    };
+
+    let token = utils::user::generate_jwt_auth_token_with_attributes(
```

Comment:
if you have time, can you refactor this function to take only `user_id` instead of user. we can save a DB call.